### ISSUE TITLE
Custom dataset preprocessing

### ISSUE THREAD

**Issue by ayukh** [erfanzar/EasyDeL#171]
Hi, thanks for the package! I have a custom non-HuggingFace dataset and I found the following documentation for [data processing](https://github.com/erfanzar/EasyDeL/blob/513fc33a16f6465e88a14cd74b2295ff6fe78a1b/docs/data_processing.md), but I see that the `from easydel.data_preprocessing import DataProcessor, DataProcessorArguments` methods are not included in EasyDeL anymore?
Is there some way to use non-HF dataset? Source code transforms dataset to Tensorflow dataset, but in my case I have an already tokenized dataset loaded using numpy.memmap for better memory management. Thanks!

**Comment by erfanzar** [comment]
hi
yes you can tweak the trainers if you want and customize that
but in case you don't want to do that, you can give me a sample of how u store data (IDs, mask) and I'll add that for you.

**Comment by ayukh** [comment]
Hi @erfanzar, thanks for the quick reply! 
So the dataset that we use has sensitive information, so I am not sure what I am allowed to share in this case. I was just confused about the documentation - I will try to configure this myself and if I have any questions I will reply here (if I don't reply in a week then I will close the issue).

Do I understand correctly that the dataset does not have to be in Tensorflow format specifically to use EasyDeL?

**Comment by erfanzar** [comment]
A dataset doesn’t have to be a TensorFlow dataset; you can create and configure your own datasets and dataloaders. You can easily utilize libraries like NumPy, PyTorch, or others to build your dataloaders.

**Comment by ayukh** [comment]
Good to know, thank you! Will experiment with this.
